internal/runners/packages/table.go

```diff
@@ -4,44 +4,69 @@ import (
 	"sort"
 	"strings"
 
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/output"
 	"github.com/bndr/gotabulate"
 )
 
-type table struct {
-	headers []string
-	data    [][]string
+type packageTable struct {
+	rows        []packageRow
+	emptyOutput string // string returned when table is empty
 }
 
-func newTable(headers []string, data [][]string) *table {
-	return &table{
-		headers: headers,
-		data:    data,
+type packageRow struct {
+	Pkg     string `json:"package"`
+	Version string `json:"version"`
+}
+
+func (t *packageTable) MarshalOutput(format output.Format) interface{} {
+	if format == output.PlainFormatName {
+		if len(t.rows) == 0 {
+			return t.emptyOutput
+		}
+		return t.output()
+	}
+
+	type packageRow struct {
+		Pkg     string `json:"package"`
+		Version string `json:"version"`
+	}
+
+	return t.rows
+}
+
+func newTable(rows []packageRow, emptyOutput string) *packageTable {
+	return &packageTable{
+		rows:        rows,
+		emptyOutput: emptyOutput,
 	}
 }
 
-func (t *table) output() string {
+func (t *packageTable) output() string {
 	if t == nil {
-		return ""
+		return t.emptyOutput
 	}
 
-	tab := gotabulate.Create(t.data)
-	tab.SetHeaders(t.headers)
+	headers := []string{
+		locale.T("package_name"),
+		locale.T("package_version"),
+	}
+
+	data := make([][]string, 0, len(t.rows))
+	for _, row := range t.rows {
+		data = append(data, []string{row.Pkg, row.Version})
+	}
+	tab := gotabulate.Create(data)
```

Comment:
We should not be using gotabulate here at all, the outputter facilitates table output.

eg. http://github.com/ActiveState/cli/blob/cf6e4835824104807fc163d8d79302b95214ffab/internal/runners/events/events.go#L27-L27

Reply:
I see.  I looked at the scripts implementation first: https://github.com/ActiveState/cli/blob/master/internal/runners/scripts/scripts.go#L45-L51  where we are still using gotabulate .

But yes, I will change it this way.  That's better for sure...

Reply:
I'll file a bug for scripts, thanks!